src/ACEtk/block/TabulatedEnergyAngleDistribution.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef NJOY_ACETK_BLOCK_TABULATEDOUTGOINGENERGYANGLEDISTRIBUTION
+#define NJOY_ACETK_BLOCK_TABULATEDOUTGOINGENERGYANGLEDISTRIBUTION
+
+// system includes
+#include <variant>
+
+// other includes
+#include "ACEtk/block/details/BaseDistributionData.hpp"
+#include "ACEtk/block/TabulatedAngularDistribution.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+
+/**
+ *  @class
+ *  @brief Tabulated outgoing energy and angle distribution data for a single
+ *         incident energy
+ *
+ *  The TabulatedEnergyAngleDistribution class contains the
+ *  tabulated angular distributions for a set of outgoing energy values and
+ *  associated incident energy value. It is used in the
+ *  OutgoingEnergyAngleDistributionData (ACE LAW 61) in the DLW block.
+ */
+class TabulatedEnergyAngleDistribution :
+  protected details::BaseDistributionData< TabulatedEnergyAngleDistribution,
+                                           TabulatedAngularDistribution > {
+
+  friend class details::BaseDistributionData< TabulatedEnergyAngleDistribution,
+                                              TabulatedAngularDistribution >;
+
+  /* fields */
+  double incident_;
+
+  /* auxiliary functions */
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/TabulatedEnergyAngleDistribution/src/ctor.hpp"
+
+  // generic function used internally
+  double value() const { return this->incident_; }
+
+  /**
+   *  @brief Return the incident energy value
+   */
+  double incidentEnergy() const { return this->value(); }
+
+  /**
+   *  @brief Return the interpolation data
+   */
+  auto interpolationData() const {
+
+    return BaseDistributionData::interpolationData();
+  }
+
+  /**
+   *  @brief Return the number of interpolation regions
+   */
+  std::size_t NB() const { return BaseDistributionData::NB(); }
+
+  /**
+   *  @brief Return the number of interpolation regions
+   */
+  std::size_t numberInterpolationRegions() const {
+
+    return BaseDistributionData::numberInterpolationRegions();
+  }
+
+  /**
+   *  @brief Return the interpolation boundaries
+   */
+  auto NBT() const { return BaseDistributionData::NBT(); }
+
+  /**
+   *  @brief Return the interpolation boundaries
+   */
+  auto boundaries() const { return BaseDistributionData::boundaries(); }
+
+  /**
+   *  @brief Return the interpolants
+   */
+  auto INT() const { return BaseDistributionData::INT(); }
+
+  /**
+   *  @brief Return the interpolants
+   */
+  auto interpolants() const { return BaseDistributionData::interpolants(); }
+
+  /**
+   *  @brief Return the number of outgoing energy values
+   */
+  std::size_t NE() const { return BaseDistributionData::N(); }
+
+  /**
+   *  @brief Return the number of outgoing energy values
+   */
+  std::size_t numberOutgoingEnergies() const {
+
+    return BaseDistributionData::numberValues();
+  }
+
+  /**
+   *  @brief Return the outgoing energy values
+   */
+  auto outgoingEnergies() const {
```

Comment:
![image](https://user-images.githubusercontent.com/4073387/159699884-4cc74a4f-7ec6-4049-bc9b-df96680fe0a0.png)

I believe outgoingEnergies needs an associated set of PDF and CDF values - see attached screenshot of the ACE manual

Reply:
You are right. I did this wrong. I don't understand how I missed this. I'll have to rewrite this class since the base class is not appropriate for this one.